django/thunderstore/account/authentication.py

```diff
@@ -0,0 +1,21 @@
+import datetime
+
+from rest_framework.authentication import TokenAuthentication as DRFTokenAuthentication
+from rest_framework.authtoken.models import Token
+
+from thunderstore.account.models import ServiceAccount
+
+
+class TokenAuthentication(DRFTokenAuthentication):
+    keyword = "Bearer"
+
+    def authenticate(self, request):
+        out = super().authenticate(request)
+        if out is not None and all(out):
+            # The request has been authenticated
+            token: Token = out[1]
+            service_account = ServiceAccount.objects.filter(user=token.user).first()
+            if service_account:
+                service_account.last_used = datetime.datetime.now()
```

Comment:
Not sure if this is the right datetime class or if a Django class/system is preferred

Reply:
> DateTimeField ServiceAccount.last_used received a naive datetime (2021-01-31 18:09:56.049794) while time zone support is active.

Reply:
Always use [timezone.now()](https://docs.djangoproject.com/en/3.1/topics/i18n/timezones/#naive-and-aware-datetime-objects) with Django